pyof/foundation/network_types.py

```diff
@@ -142,7 +142,7 @@ def __init__(self, destination=None, source=None, vlan=VLAN(),
         self.source = source
         self.vlan = vlan
         self.ether_type = ether_type
-        self.data = data
+        self.data = b'' if data is None else data
```

Comment:
The cleaner way is to set `b''` directly in the `__init__` method - same behavior, better readability.